### PR TITLE
Added grat email tests and refactored

### PR DESCRIPTION
**Pull request by camhashemi**
@andrewconner

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/model/GratificationData.scala [hunk 1]
```diff
@@ -2,29 +2,32 @@ package com.keepit.model
...
+import play.api.libs.json._
 
-@json
-case class LibraryCountData(totalCount: Int, countById: Map[Id[Library], Int]) {
-
-  val sortedCountById = countById.toList.sortWith { _._2 > _._2 }
+case class CountData[T](totalCount: Int, countById: Map[Id[T], Int]) {
+  val sortedCountById = countById.toList.sortWith {
+    _._2 > _._2
+  }
```
Comment (yingjieMiao) [inline kifi-backend/modules/common/app/com/keepit/model/GratificationData.scala]: `countById.toArray.sortBy(-_._2)` ?

Comment (camhashemi) [inline kifi-backend/modules/common/app/com/keepit/model/GratificationData.scala]: oo pretty. thank you.

File: kifi-backend/modules/common/app/com/keepit/model/GratificationData.scala [hunk 2]
```diff
@@ -2,29 +2,30 @@ package com.keepit.model
...
 import com.kifi.macros.json
+import play.api.libs.functional.syntax._
+import play.api.libs.json._
 
-@json
-case class LibraryCountData(totalCount: Int, countById: Map[Id[Library], Int]) {
-
-  val sortedCountById = countById.toList.sortWith { _._2 > _._2 }
+case class CountData[T](totalCount: Int, countById: Map[Id[T], Int]) {
+  val sortedCountById = countById.toList.sortBy(-_._2)
```
Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/model/GratificationData.scala]: Make it a def and you save yourself having to sort until you need it. If you plan on calling it more than once, `lazy val`

**Comment by andrewconner** [comment]
Cool abstraction (higher kinded type). Looks good, sans the `val` comment

**Pull request merged**